core/tests/snapshots.rs

```diff
@@ -370,8 +369,24 @@ mod tests {
 
             bank_forks.insert(bank);
             if slot == saved_slot as u64 {
-                let options = CopyOptions::new();
-                fs_extra::dir::copy(accounts_dir, &saved_accounts_dir, &options).unwrap();
+                // Find the relevant snapshot storages
+                let snapshot_storage_files: HashSet<_> = bank_forks[slot]
+                    .get_snapshot_storages()
+                    .into_iter()
+                    .flatten()
+                    .map(|s| s.get_path())
+                    .collect();
+
+                // Only save off the files returned by `get_snapshot_storages`. This is because
+                // some of the storage entries in the accounts directory may be filtered out by
+                // `get_snapshot_storages()` and will not be included in the snapshot.
```

Comment:
nits: Add this? `// So, copying naitvely everything in accounts_dir could lead to test failure by mismatch`